src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Tests/UnitTests/Controllers/TestControllerTests.cs

```diff
@@ -0,0 +1,188 @@
+            Assert.IsType<TestDto>(viewResult.ViewData.Model);
+        }
+
+        [Fact]
+        public void Index_TestEnded_ReturnsViewWithError()
+        {
+            // Arrange
+            var test = GetTest(TestStatus.Ended);
+            var answer = "Answer";
+            var user = GetUser();
+            this._testServiceMock.Setup(x => x.GetTestByNumber(test.Id)).Returns(test);
+            using var controller = new TestController(this._testServiceMock.Object);
+            controller.ControllerContext = new ControllerContext()
+            {
+                HttpContext = new DefaultHttpContext() { User = user }
+            };
+
+            // Act
+            var result = controller.Index(test.Id, answer);
+
+            // Assert
+            var allErrors = controller.ModelState.Values.SelectMany(v => v.Errors);
+            Assert.Single(allErrors);
+            Assert.Contains(allErrors, x => x.ErrorMessage == ExceptionsMessages.ErrorOccurs);
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.IsType<TestDto>(viewResult.ViewData.Model);
+        }
+
+        [Fact]
+        public void Index_WrongAnswer_ReturnsViewWithError()
+        {
+            // Arrange
+            var test = GetTest(TestStatus.Started);
+            var answer = "WrongAnswer";
+            var user = GetUser();
+            this._testServiceMock.Setup(x => x.GetTestByNumber(test.Id)).Returns(test);
+            using var controller = new TestController(this._testServiceMock.Object);
+            controller.ControllerContext = new ControllerContext()
+            {
+                HttpContext = new DefaultHttpContext() { User = user }
+            };
+
+            // Act
+            var result = controller.Index(test.Id, answer);
+
+            // Assert
+            var allErrors = controller.ModelState.Values.SelectMany(v => v.Errors);
+            Assert.Single(allErrors);
+            Assert.Contains(allErrors, x => x.ErrorMessage == ExceptionsMessages.ErrorTryAgain);
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.IsType<TestDto>(viewResult.ViewData.Model);
+        }
+
+        [Fact]
+        public void Index_CorrectAnswer_ReturnsAnswerViewModel()
+        {
+            // Arrange
+            var test = GetTest(TestStatus.Started);
+            var answer = "Answer1";
+            var user = GetUser();
+            this._testServiceMock.Setup(x => x.GetTestByNumber(test.Id)).Returns(test);
+            this._testServiceMock.Setup(x => x.VerifyTestAnswer(It.IsAny<string>(), It.IsAny<List<string>>())).Returns(true);
+            using var controller = new TestController(this._testServiceMock.Object);
+            controller.ControllerContext = new ControllerContext()
+            {
+                HttpContext = new DefaultHttpContext() { User = user }
+            };
+
+            // Act
+            var result = controller.Index(test.Id, answer);
+        }
+
```

Comment:
I had toubles with that Act and Assert. Dont know how to mock var answerViewModel = this.SaveAnswer(test);